Sources/NostrSDK/Events/Channels/CreateChannelMessageEvent.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  CreateChannelMessageEvent.swift
+//
+//
+//  Created by Konstantin Yurchenko, Jr on 9/11/24.
+//
+
+import Foundation
+
+/// Send a text message to a channel.
+/// See [NIP-28](https://github.com/nostr-protocol/nips/blob/master/28.md#kind-42-create-channel-message).
+public class CreateChannelMessageEvent: NostrEvent {
+    
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+    
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    required init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    required init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), pubkey: String) {
+        super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, pubkey: pubkey)
+    }
+
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(id: String, pubkey: String, createdAt: Int64, kind: EventKind, tags: [Tag], content: String, signature: String?) {
+        super.init(id: id, pubkey: pubkey, createdAt: createdAt, kind: kind, tags: tags, content: content, signature: signature)
+    }
+
+    init(content: String, tags: [Tag], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: Self.kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+    
+    class var kind: EventKind {
+        .channelMessage
+    }
+}
+
+public extension EventCreating {
+    func createChannelMessageEvent(withContent content: String, eventId: String, relayUrl: String, signedBy keypair: Keypair) throws -> CreateChannelMessageEvent {
```

Comment:
Same here. Again, look at how TextNoteEvent's Builder handles NIP-10 marker event tags. Maybe you could refactor or at least follow what we do in that code.

NIP-28 says that root or reply event tags are supported.

Reply:
Yes, I will take care of this when I am working on replies in my app.

Reply:
FYI I'm working on [NIP-17 - Private Direct Messages](https://github.com/nostr-protocol/nips/blob/master/17.md) which requires threading as well, so I ended up refactoring the logic for building events with NIP-10 tags. It hasn't been merged yet -- still working on my branch but it'll be ready soon. Hope you haven't gotten to it yet, don't want to make your changes redundant.

Reply:
Great! Thank you so much.

Reply:
Merged: https://github.com/nostr-sdk/nostr-sdk-ios/pull/186